dds/DCPS/InternalDataReader.h

```diff
@@ -28,31 +30,65 @@ namespace DCPS {
 class InternalEntity : public virtual RcObject {};
 typedef WeakRcHandle<InternalEntity> InternalEntity_wrch;
 
-enum InternalSampleInfoKind {
-  ISIK_REGISTER,
-  ISIK_SAMPLE,
-  ISIK_UNREGISTER,
-  ISIK_DISPOSE
-};
-
-struct InternalSampleInfo {
-  InternalSampleInfoKind kind;
-  InternalEntity_wrch publication_handle;
-
-  InternalSampleInfo(InternalSampleInfoKind a_kind,
-                     InternalEntity_wrch a_publication_handle)
-    : kind(a_kind)
-    , publication_handle(a_publication_handle)
+typedef OPENDDS_VECTOR(DDS::SampleInfo) InternalSampleInfoSequence;
+
+inline DDS::SampleInfo make_sample_info(DDS::SampleStateKind sample_state,
+                                        DDS::ViewStateKind view_state,
+                                        DDS::InstanceStateKind instance_state,
+                                        size_t disposed_generation_count,
+                                        size_t no_writers_generation_count,
+                                        size_t sample_rank,
+                                        size_t generation_rank,
+                                        size_t absolute_generation_rank,
+                                        bool valid_data)
+{
+  DDS::SampleInfo si;
+  si.sample_state = sample_state;
+  si.view_state = view_state;
+  si.instance_state = instance_state;
+  si.source_timestamp = make_time_t(0, 0); // TODO
+  si.instance_handle = DDS::HANDLE_NIL; // TODO
+  si.publication_handle = DDS::HANDLE_NIL; // TODO
+  si.disposed_generation_count = disposed_generation_count;
+  si.no_writers_generation_count = no_writers_generation_count;
+  si.sample_rank = sample_rank;
+  si.generation_rank = generation_rank;
+  si.absolute_generation_rank = absolute_generation_rank;
+  si.valid_data = valid_data;
+  return si;
+}
+
+#ifndef OPENDDS_SAFETY_PROFILE
+inline bool operator==(const DDS::SampleInfo& x, const DDS::SampleInfo& y)
+{
+  return x.sample_state == y.sample_state &&
+    x.view_state == y.view_state &&
+    x.instance_state == y.instance_state &&
+    x.source_timestamp == y.source_timestamp &&
+    x.instance_handle == y.instance_handle &&
+    x.publication_handle == y.publication_handle &&
+    x.disposed_generation_count == y.disposed_generation_count &&
+    x.no_writers_generation_count == y.no_writers_generation_count &&
+    x.sample_rank == y.sample_rank &&
+    x.generation_rank == y.generation_rank &&
+    x.absolute_generation_rank == y.absolute_generation_rank &&
+    x.valid_data == y.valid_data;
+}
+#endif
+
+class SIW {
```

Comment:
This seems a bit too terse. Also I don't know how extensive this header is, but a short all caps identifier like this seems vulnerable to be messed by the predecessor. What about `SIWrap`?